### ISSUE TITLE
Dropdown button has invalid ARIA attribute (aria-invalid)

### ISSUE THREAD

**Issue by maggiewachs** [CMSgov/design-system#3290]
**Describe the bug**
The dropdown component contains a button with `aria-invalid` assigned, which is being flagged by our instance of Siteimprove as an invalid use of ARIA. Dropdown source: https://github.com/CMSgov/design-system/blob/main/packages/design-system/src/components/Dropdown/Dropdown.tsx#L279

In previous ARIA recommendations, `aria-invalid` had a global scope, but the most recent version (1.2), [narrowed the scope to specific roles](https://www.w3.org/TR/wai-aria-1.2/#aria-invalid). The aria-invalid attribute should be reassigned to the `role="listbox"` element.

**To Reproduce**
1. Go to [the Dropdown](https://design.cms.gov/components/dropdown/?theme=core)
2. Inspect the dropdown example's source, and notice how the `<button>` element has `aria-invalid="false"`
3. Close the inspector, and run the [Siteimprove browser extension](https://www.siteimprove.com/integrations/browser-extensions/)
4. Siteimprove reports the error, "ARIA attribute unsupported or prohibited", and references `aria-invalid` on the button

**Expected behavior**
No ARIA violations.

**Desktop (please complete the following information):**
 - OS: Mac OS 14.6.1
 - Browser: Chrome latest
 - Version: CMS DS v11
